lib/src/game/game_core.dart

```diff
@@ -151,7 +151,7 @@ class Game {
                 char = '?';
                 break;
               default:
-              // noop;
+                throw StateError('Got into a bad state!');
```

Comment:
Caught by DDC, but not analyzer!

Reply:
Analyzer does yell at you, when you attempt to use `char` that is not definitely assigned if `default` does not throw (completes).

OTOH, it is OK to have `default` and do nothing, and without `default` you will get a warning that `switch` does not handle all enum values. From the spec: "If T is an enum type, it is a warning if the switch does not handle all enum cases, either explicitly or via a default."

Reply:
I'm arguing as a naive developer. `dartanalyzer` doesn't mind my code, but DDC wanted a throw here or `char` is not assigned.

A warning or hint from analyzer would be good!

Reply:
@kevmoo can you clarify?  Are you saying that if you delete the `throw StateError` the analyzer does not complain?  If so, that's a bug, I think.  Or am I missing something @scheglov ?

Reply:
The analyzer DOES NOT complain, but the CFE does! – I thought it was a bug, too!

Reply:
@leafpetersen and @kevmoo  What kind of "complain" would you expect here? If I comment out the `throw` in `default`, there is no diagnostic for the `switch`. But because now `char` is not definitely assigned, there is an error below, right after the `switch`, where we have `assert(char != null);  buffer.write(char);`.

Reply:
I'd expect a hint on `char != null` too! (Since it can never be null – it's explicit type is `String char`.

I'd assume analyzer should warn at 158 or 159 – it doesn't!

Reply:
Works for me, with the bleeding edge analyzer.
![image](https://user-images.githubusercontent.com/384794/75467947-4561b900-5941-11ea-84fb-c44317af2f6a.png)

Reply:
Just confirmed on edge myself! Awesome!

Reply:
👍